src/client.h

```diff
@@ -200,16 +200,21 @@ class Client : public ClientBase {
     }
 
     inline unsigned int client_receive_from_selected(int ifd) {
-#ifdef USING_VMA_EXTRA_API
-        if (SOCKETXTREME == g_pApp->m_const_params.fd_handler_type) {
+#ifdef USING_VMA_EXTRA_API // VMA
+        if (SOCKETXTREME == g_pApp->m_const_params.fd_handler_type && g_vma_api) {
             return client_receive_from_selected<SocketXtremeInputHandler>(ifd);
-        } else if (g_pApp->m_const_params.is_zcopyread) {
+        }
+
+        if (g_pApp->m_const_params.is_zcopyread && g_vma_api) {
             return client_receive_from_selected<VmaZCopyReadInputHandler>(ifd);
-        } else
+        }
 #endif
-        {
-            return client_receive_from_selected<RecvFromInputHandler>(ifd);
+#ifdef USING_XLIO_EXTRA_API // XLIO
+        if (g_pApp->m_const_params.is_zcopyread && g_xlio_api) {
+            return client_receive_from_selected<XlioZCopyReadInputHandler>(ifd);
         }
+#endif
```

Comment:
It's difficulot to locate the corresponding #ifdef in diff mode. So please add comments to #endifs.
```suggestion
#endif // USING_XLIO_EXTRA_API 
```

Reply:
Done